### DISCUSSION TITLE
why findOrCreate in the readme example?

### DISCUSSION THREAD

**Discussion by fbialek** [sergiodxa/remix-auth-form#30] [Q&A] [answered]
Maybe this is a stupid question as this is about an example - but why in the readme example there is following snippet:

```
    // And finally, you can find, or create, the user
    let user = await findOrCreateUser(username, hashedPassword);
```

I would assume that in most cases you will want only to find the user while authenticating. 
Are there some use cases of creating users in the auth process?

**Comment by sergiodxa** [comment, marked as answer]
If you use the same strategy instance for both login and signup you need to be able to do a find or create.

You can also create a different strategy instance for each form and there only do a find or a create depending if it's login or signup.
